consumers/campaigner/main.go

```diff
@@ -243,12 +246,58 @@ func processSubscribers(
 	return nil
 }
 
-func setStatusFailed(ctx context.Context, store storage.Storage, campaign *entities.Campaign) error {
+// overwriting the callback func for max attempts reached to insert into campaign failed logs.
+func (h *MessageHandler) LogFailedMessage(m *nsq.Message) {
+	if m == nil {
+		return
+	}
+	msg := new(entities.CampaignerTopicParams)
+	err := json.Unmarshal(m.Body, &msg)
+	if err != nil {
+		logrus.WithField("body", string(m.Body)).
+			WithError(err).Error("Malformed JSON message.")
+		return
+	}
+	logrus.WithFields(logrus.Fields{
+		"user_id":     msg.UserID,
+		"campaign_id": msg.CampaignID,
+	}).Error("Exceeded max attempts for sending the e-mail.")
+
+	log := &entities.CampaignFailedLog{
+		ID:          ksuid.New(),
+		UserID:      msg.UserID,
+		CampaignID:  msg.CampaignID,
+		Description: "Exceeded max attempts for preparing the campaign",
+		CreatedAt:   time.Time{},
+	}
+	campaign, err := storage.GetCampaign(context.Background(), msg.CampaignID, msg.UserID)
+	if err != nil {
+		logrus.WithField("campaign_id", msg.CampaignID).
+			WithError(err).Error("Failed to get campaign.")
+		return
+	}
+	err = storage.LogFailedCampaign(context.Background(), campaign, log)
+	if err != nil {
+		logrus.WithField("log", log).
+			WithError(err).Error("Failed to store campaign failed log.")
+		return
+	}
+}
+
+// logFailedCampaign updates campaign status to failed & inserts campaign  failed log.
+func logFailedCampaign(ctx context.Context, store storage.Storage, campaign *entities.Campaign) error {
 	defer trace.StartRegion(ctx, "setStatusFailed").End()
 
 	campaign.Status = entities.StatusFailed
 	campaign.CompletedAt.SetValid(time.Now().UTC())
-	return store.UpdateCampaign(campaign)
+	log := &entities.CampaignFailedLog{
+		ID:          ksuid.New(),
+		UserID:      campaign.UserID,
+		CampaignID:  campaign.ID,
+		Description: "Failed to prepare campaign",
```

Comment:
The description depends on where we invoke the `logFailedCampaign` func. I think we should add it as an argument in the function.